packages/core/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import { GlintConfig, loadConfig } from '@glint/config';
+import DocumentCache from './common/document-cache';
+import TransformManager from './common/transform-manager';
+import GlintLanguageServer from './language-server/glint-language-server';
+
+/** @internal */
+export interface ProjectAnalysis {
+  glintConfig: GlintConfig;
+  transformManager: TransformManager;
+  languageServer: GlintLanguageServer;
+  shutdown: () => void;
+}
+
+/**
+ * This function is available to consumers as an unstable API. We will not go
+ * out of our way to change or break it, but there may be breaking changes
+ * to its behavior or type signature outside of major version bumps.
```

Comment:
This is the (TypeScript) way.

<img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia1.tenor.com%2Fimages%2Fdeb73ac86e6db73efa309d861aad597d%2Ftenor.gif%3Fitemid%3D15956069&f=1&nofb=1&ipt=d2faf642c2aa422b814f6fc4bc31a6bd0ce60157d7591eb9356ba8c48746fba2&ipo=images">